.github/workflows/publishing.yml

```
@@ -158,6 +160,54 @@ jobs:
         env:
           GH_TOKEN: ${{ secrets.REPO_BOT_ACCESS_TOKEN }}
 
+      - name: Retrieve wheel assets
+        run: |
+          assets_folder=/tmp/wheel_assets && mkdir "$assets_folder" && cd "$assets_folder"
+          
+          function download_assets {
+            mkdir "$1" && cd "$1"
+            release_id=`echo "$2" | jq -r ".$1.release_id"`
+            echo "Download $release_id assets..."
+            gh release download $release_id -R ${{ github.repository }}
+            cd ..
+          }
+          
+          for file in ${{ join(fromJson(steps.artifacts.outputs.python_wheels).info_files, ' ') }}; do
+            download_assets $file '${{ steps.artifacts.outputs.python_wheels }}'
+          done
+         
+        env:
+          GH_TOKEN: ${{ secrets.REPO_BOT_ACCESS_TOKEN }}
+  
+      - name: Upload wheel assets
+        uses: actions/upload-artifact@v3
+        with:
+          name: wheel_assets
+          path: /tmp/wheel_assets
+          retention-days: 1
+          if-no-files-found: error    
+      
+      # We can delete staging branch now after downloading the assets (as artifacts for later wheel jobs). 
+      - name: Clean up
+        run: |
+          # Clean up the staging branch that was used to trigger the GitLab pipeline.
+          git config --global user.name "cuda-quantum-bot"
+          git config --global user.email "cuda-quantum-bot@users.noreply.github.com"
+          function delete_staging_branch {
+            staging_branch=`echo "$2" | jq -r ".$1.staging_branch"`
+            echo "Delete staging branch $staging_branch ..."
+            if [[ "$staging_branch" =~ ^bot\/.*$ ]]; then
+              git push origin --delete $staging_branch || true
```

I'd prefer to not silently ignore a failure to delete the branch unless we did not create a staging branch
```suggestion
              git push origin --delete $staging_branch || ${{ inputs.assets_from_run != '' }}
```
You could also give "Trigger assets creation" and id and check if the step was skipped.

---

Fixed as suggested.